clash-prelude/src/Clash/XException.hs

```diff
@@ -6,14 +6,58 @@ Copyright  :  (C) 2016,      University of Twente,
 License    :  BSD2 (see the file LICENSE)
 Maintainer :  QBayLogic B.V. <devops@qbaylogic.com>
 
-'XException': An exception for uninitialized values
+This module houses utilities to deal with /undefined/ values in Clash. An undefined
+value is translated to @x@ in VHDL and Verilog. At its core, this is handled by
+a special 'Exception' called 'XException'.
 
->>> show (errorX "undefined" :: Integer, 4 :: Int)
-"(*** Exception: X: undefined
-CallStack (from HasCallStack):
+Like any 'Exception' in Haskell, any term can take on this value:
+
+>>> error "help" :: Int
+*** Exception: help
+...
+>>> errorX "help" :: Int
+*** Exception: X: help
 ...
->>> showX (errorX "undefined" :: Integer, 4 :: Int)
-"(undefined,4)"
+
+Within the Haskell ecosystem, and throughout this module, you will find these
```

Comment:
I wonder if it's a good idea to mention bottom at all. If I was unfamiliar with the concept I would now be confused about what it means. Attempting to look up it's definition may lead to PLT type definitions which only add more confusion

Reply:
Yeah if we don't we'll have to change existing documentation. But that's fine.